lapack/unit_test/Test_Lapack_svd.hpp

```diff
@@ -0,0 +1,321 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+#include <KokkosKernels_TestUtils.hpp>
+#include <KokkosBlas3_gemm.hpp>
+
+#include <KokkosLapack_svd.hpp>
+
+namespace Test {
+
+template <class AMatrix, class SVector, class UMatrix, class VMatrix>
+void check_triple_product(const AMatrix& A, const SVector& S, const UMatrix& U,
+                          const VMatrix& Vt) {
+  // After a successful SVD decomposition we have A=U*S*V
+  // So using gemm we should be able to compare the above
+  // triple product to the original matrix A.
+
+  using scalar_type = typename AMatrix::non_const_value_type;
+  using mag_type    = typename Kokkos::ArithTraits<scalar_type>::mag_type;
+
+  AMatrix temp("intermediate U*S product", A.extent(0), A.extent(1));
+  AMatrix M("U*S*V product", A.extent(0), A.extent(1));
+
+  // First compute the left side of the product: temp = U*S
+  KokkosBlas::gemm("N", "N", 1, U, S, 0, temp);
+
+  // Second compute the right side of the product: M = temp*V = U*S*V
+  KokkosBlas::gemm("N", "C", 1, temp, Vt, 0, M);
+
+  typename AMatrix::HostMirror A_h = Kokkos::create_mirror_view(A);
+  typename AMatrix::HostMirror M_h = Kokkos::create_mirror_view(M);
+  Kokkos::deep_copy(A_h, A);
+  Kokkos::deep_copy(M_h, M);
+  constexpr mag_type tol = 100 * Kokkos::ArithTraits<scalar_type>::eps();
+  for (int rowIdx = 0; rowIdx < A.extent_int(0); ++rowIdx) {
+    for (int colIdx = 0; colIdx < A.extent_int(1); ++colIdx) {
+      EXPECT_NEAR_KK_REL(A_h(rowIdx, colIdx), M_h(rowIdx, colIdx), tol);
+    }
+  }
+}
+
+template <class Matrix>
+void check_unitary_orthogonal_matrix(const Matrix& M) {
```

Comment:
Same as above?

Reply:
Same as above, will be used to check answers from random matrix test